conans/client/source.py

```diff
@@ -5,15 +5,31 @@
 from conans.client.cmd.export import export_recipe, export_source
 from conans.client.tools import no_op
 from conans.errors import ConanException, ConanExceptionInUserConanfileMethod, \
-    conanfile_exception_formatter
+    conanfile_exception_formatter, NotFoundException
 from conans.model.scm import SCM, get_scm_data
 from conans.paths import CONANFILE, CONAN_MANIFEST, EXPORT_SOURCES_TGZ_NAME, EXPORT_TGZ_NAME
 from conans.util.conan_v2_mode import conan_v2_property
 from conans.util.files import (is_dirty, mkdir, rmdir, set_dirty_context_manager,
                                merge_directories, clean_dirty)
 
 
-def retrieve_exports_sources(remote_manager, cache, recipe_layout, conanfile, ref, remotes):
+def _try_get_sources(ref, remote_manager, recipe_layout, remote):
+    try:
+        remote_manager.get_recipe_revisions(ref, remote)
```

Comment:
why the double remote_manager call? Can't we call just ``get_recipe_sources`` and filter on return exception?